setup.py

```diff
@@ -72,7 +72,7 @@ def install_needed_req(import_name, package_name=None, min_version=None, max_ver
 
 setup_requirements = common_requirements + [
     'scikit-build>=0.11.0',
-    'cmake!=3.17,!=3.17.0',
```

Comment:
I am unsure why cmake is installed again here. Do you sometimes not use the pyproject.toml or is this stillonly here for legacy reasons? Commented out for testing.

Reply:
@RSwarnkar this fixes the issue. However it still fails to install scipy which is a dependency. This again requires meson-python which requires ninja, which still fails to build.

So I guess we need to ask meson-python to make ninja optional as well

Reply:
Yeah, cmake is installed via `pkg`. Here is version info: `cmake version 3.32.2`.  Termux version `0.118.0`

- Also, `pip install ninja` was throwing error too which was resolved after referring [this](https://github.com/termux/termux-packages/issues/8951#issuecomment-1207196457). 
- Pl. check [this](https://github.com/termux/termux-packages/issues/10065#issuecomment-1207254645) as well.

Reply:
Full log: 
```
pip install git+https://github.com/maxbachmann/qiskit-aer.git@09e69c90de035d7b86b278f65d38859d41d7832aCollecting git+https://github.com/maxbachmann/qiskit-aer.git@09e69c90de035d7b86b278f65d38859d41d7832a
  Cloning https://github.com/maxbachmann/qiskit-aer.git (to revision 09e69c90de035d7b86b278f65d38859d41d7832a) to /data/data/com.termux/files/usr/tmp/pip-req-build-gjsdusus
  Running command git clone --filter=blob:none --quiet https://github.com/maxbachmann/qiskit-aer.git /data/data/com.termux/files/usr/tmp/pip-req-build-gjsdusus
  Running command git rev-parse -q --verify 'sha^09e69c90de035d7b86b278f65d38859d41d7832a'
  Running command git fetch -q https://github.com/maxbachmann/qiskit-aer.git 09e69c90de035d7b86b278f65d38859d41d7832a
  Resolved https://github.com/maxbachmann/qiskit-aer.git to commit 09e69c90de035d7b86b278f65d38859d41d7832a
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting qiskit-terra>=0.20.0
  Using cached qiskit-terra-0.21.1.tar.gz (6.6 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy>=1.16.3
  Downloading numpy-1.23.2.tar.gz (10.7 MB)
     ━━━━━━━━━━━━━━ 10.7/10.7 MB 2.4 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting scipy>=1.0
  Downloading scipy-1.9.0.tar.gz (42.0 MB)
     ━━━━━━━━━━━━━━ 42.0/42.0 MB 2.1 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [61 lines of output]
      The Meson build system
      Version: 0.62.2
      Source dir: /data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619
      Build dir: /data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619/.mesonpy-5dh3m2j8/build
      Build type: native build
      Project name: SciPy
      Project version: 1.9.0
      C compiler for the host machine: cc (clang 14.0.6 "clang version 14.0.6")
      C linker for the host machine: cc ld.lld 14.0.6
      C++ compiler for the host machine: c++ (clang 14.0.6 "clang version 14.0.6")
      C++ linker for the host machine: c++ ld.lld 14.0.6
      Host machine cpu family: aarch64
      Host machine cpu: aarch64
      Compiler for C supports arguments -Wno-unused-but-set-variable: YES
      Library m found: YES

      ../../meson.build:41:0: ERROR: Unknown compiler(s): [['gfortran'], ['flang'], ['nvfortran'], ['pgfortran'], ['ifort'], ['g95']]
      The following exception(s) were encountered:
      Running "gfortran --version" gave "[Errno 2] No such file or directory: 'gfortran'"
      Running "gfortran -V" gave "[Errno 2] No such file or directory: 'gfortran'"
      Running "flang --version" gave "[Errno 2] No such file or directory: 'flang'"
      Running "flang -V" gave "[Errno 2] No such file or directory: 'flang'"
      Running "nvfortran --version" gave "[Errno 2] No such file or directory: 'nvfortran'"
      Running "nvfortran -V" gave "[Errno 2] No such file or directory: 'nvfortran'"
      Running "pgfortran --version" gave "[Errno 2] No such file or directory: 'pgfortran'"
      Running "pgfortran -V" gave "[Errno 2] No such file or directory: 'pgfortran'"
      Running "ifort --version" gave "[Errno 2] No such file or directory: 'ifort'"
      Running "ifort -V" gave "[Errno 2] No such file or directory: 'ifort'"
      Running "g95 --version" gave "[Errno 2] No such file or directory: 'g95'"
      Running "g95 -V" gave "[Errno 2] No such file or directory: 'g95'"

      A full log can be found at /data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619/.mesonpy-5dh3m2j8/build/meson-logs/meson-log.txt
      + meson setup --native-file=/data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619/.mesonpy-native-file.ini -Ddebug=false -Doptimization=2 --prefix=/data/data/com.termux/files/usr /data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619 /data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619/.mesonpy-5dh3m2j8/build
      Traceback (most recent call last):
        File "/data/data/com.termux/files/usr/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/data/data/com.termux/files/usr/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/data/data/com.termux/files/usr/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-utrpvz7e/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 909, in get_requires_for_build_wheel
          with _project(config_settings) as project:
        File "/data/data/com.termux/files/usr/lib/python3.10/contextlib.py", line 135, in __enter__
          return next(self.gen)
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-utrpvz7e/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 888, in _project
          with Project.with_temp_working_dir(
        File "/data/data/com.termux/files/usr/lib/python3.10/contextlib.py", line 135, in __enter__
          return next(self.gen)
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-utrpvz7e/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 547, in with_temp_working_dir
          yield cls(source_dir, tmpdir, build_dir)
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-utrpvz7e/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 463, in __init__
          self._configure(reconfigure=bool(build_dir) and not native_file_mismatch)
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-utrpvz7e/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 494, in _configure
          self._meson(
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-utrpvz7e/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 477, in _meson
          return self._proc('meson', *args)
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-utrpvz7e/overlay/lib/python3.10/site-packages/mesonpy/__init__.py", line 472, in _proc
          subprocess.check_call(list(args))
        File "/data/data/com.termux/files/usr/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['meson', 'setup', '--native-file=/data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619/.mesonpy-native-file.ini', '-Ddebug=false', '-Doptimization=2', '--prefix=/data/data/com.termux/files/usr', '/data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619', '/data/data/com.termux/files/usr/tmp/pip-install-k3_eufb4/scipy_e695a5119ee347e588c06bbe9c52b619/.mesonpy-5dh3m2j8/build']' returned non-zero exit status 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
~ $
```

Reply:
Hm as far as I can see termux does not provide a fortran compiler. It appears there are plans to add flang, but it failed to build so far: https://github.com/termux/termux-packages/pull/7910

Reply:
https://github.com/termux-user-repository/tur/blob/master/disabled-tur/python-scipy/build.sh

We can try this build.sh file for building scipy